org.eclipse.jgit.test/tst/org/eclipse/jgit/lib/ConfigTest.java

```diff
@@ -269,14 +271,22 @@ public void test007_readUserConfig() {
 
 	@Test
 	public void testReadUserConfigWithInvalidCharactersStripped() {
+		String authorName = null;
+		try {
+			authorName = SystemReader.getInstance().getenv(Constants.GIT_AUTHOR_NAME_KEY);
+		} catch (AccessControlException e) {
+			// do nothing;
+		}
+
+		Assume.assumeTrue(authorName == null);
```

Comment:
Why not have an assert here? Assume will skip the test if this condition fails.

Reply:
When env GIT_AUTHOR_NAME_KEY is specified, userConfig.getAuthorName() gets the name from this env instead of from the config that is set through the test case. This statement makes sure that the test case is executed only when env GIT_AUTHOR_NAME_KEY is not specified.